From 4d44875948eae72442c12a9ac23817c116e21a6c Mon Sep 17 00:00:00 2001
Message-ID: <4d44875948eae72442c12a9ac23817c116e21a6c.1705743627.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Jan 2024 23:14:04 +0000
Subject: [PATCH 12/38] gnu: go-gitlab-com-yawning-edwards25519-extra: Move to
 (gnu packages golang-crypto).

* gnu/packages/golang.scm (go-gitlab-com-yawning-edwards25519-extra):
Move from here ...
* gnu/packages/golang-crypto.scm: ... to here.

Change-Id: I9c4aef2a0b58c482f838cc6a3a9e443dc05629b7
---
 gnu/packages/golang-crypto.scm | 66 ++++++++++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 65 ---------------------------------
 2 files changed, 66 insertions(+), 65 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index 72771d40a2..7f25cd12a2 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -294,6 +294,72 @@ (define-public go-github-com-shadowsocks-go-shadowsocks2
 tunnel proxy protocol.")
     (license license:asl2.0)))
 
+(define-public go-gitlab-com-yawning-edwards25519-extra
+  (let ((commit "2149dcafc266f66d2487f45b156f6397f9c4760b")
+        (revision "0"))
+    (package
+      (name "go-gitlab-com-yawning-edwards25519-extra")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://gitlab.com/yawning/edwards25519-extra")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "08mz1qyi8ig515hh5blnzxhiwsav564ah7mzyhvmr6i48ndhhv98"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:unpack-path "gitlab.com/yawning/edwards25519-extra"
+         #:phases
+         (modify-phases %standard-phases
+           (replace 'build
+             (lambda arguments
+               (for-each
+                (lambda (directory)
+                  (apply (assoc-ref %standard-phases 'build)
+                         `(,@arguments #:import-path ,directory)))
+                (list
+                 "gitlab.com/yawning/edwards25519-extra/elligator2"
+                 "gitlab.com/yawning/edwards25519-extra/h2c"
+                 "gitlab.com/yawning/edwards25519-extra/internal/montgomery"
+                 "gitlab.com/yawning/edwards25519-extra/vrf"))))
+           (replace 'check
+             (lambda arguments
+               (for-each
+                (lambda (directory)
+                  (apply (assoc-ref %standard-phases 'check)
+                         `(,@arguments #:import-path ,directory)))
+                (list
+                 "gitlab.com/yawning/edwards25519-extra/elligator2"
+                 "gitlab.com/yawning/edwards25519-extra/h2c"
+                 "gitlab.com/yawning/edwards25519-extra/internal/montgomery"
+                 "gitlab.com/yawning/edwards25519-extra/vrf"))))
+           (replace 'install
+             (lambda arguments
+               (for-each
+                (lambda (directory)
+                  (apply (assoc-ref %standard-phases 'install)
+                         `(,@arguments #:import-path ,directory)))
+                (list
+                 "gitlab.com/yawning/edwards25519-extra/elligator2"
+                 "gitlab.com/yawning/edwards25519-extra/h2c"
+                 "gitlab.com/yawning/edwards25519-extra/internal/montgomery"
+                 "gitlab.com/yawning/edwards25519-extra/vrf")))))))
+      (propagated-inputs (list go-golang-org-x-crypto
+                               go-filippo-io-edwards25519))
+      (home-page "https://gitlab.com/yawning/edwards25519-extra")
+      (synopsis "edwards25519-extra")
+      (description "This package provides extensions to the Go standard
+library's Ed25519 and curve25519 implementations, primarily extracted from
+@@url{https://github.com/oasisprotocol/curve25519-voi,curve25519-voi}.  This
+package is intended for interoperability with the standard library and the
+@@url{https://filippo.io/edwards25519,edwards25519} package as much as
+possible.")
+      (license license:bsd-3))))
+
+
 ;;;
 ;;; Avoid adding new packages to the end of this file. To reduce the chances
 ;;; of a merge conflict, place them above by existing packages with similar
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 4e44ebf5ba..499bf6c8a9 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -9792,71 +9792,6 @@ (define-public age-keygen
        #:unpack-path "filippo.io/age"
        #:install-source? #f))))
 
-(define-public go-gitlab-com-yawning-edwards25519-extra
-  (let ((commit "2149dcafc266f66d2487f45b156f6397f9c4760b")
-        (revision "0"))
-    (package
-      (name "go-gitlab-com-yawning-edwards25519-extra")
-      (version (git-version "0.0.0" revision commit))
-      (source
-       (origin
-         (method git-fetch)
-         (uri (git-reference
-               (url "https://gitlab.com/yawning/edwards25519-extra")
-               (commit commit)))
-         (file-name (git-file-name name version))
-         (sha256
-          (base32 "08mz1qyi8ig515hh5blnzxhiwsav564ah7mzyhvmr6i48ndhhv98"))))
-      (build-system go-build-system)
-      (arguments
-       '(#:unpack-path "gitlab.com/yawning/edwards25519-extra"
-         #:phases
-         (modify-phases %standard-phases
-           (replace 'build
-             (lambda arguments
-               (for-each
-                (lambda (directory)
-                  (apply (assoc-ref %standard-phases 'build)
-                         `(,@arguments #:import-path ,directory)))
-                (list
-                 "gitlab.com/yawning/edwards25519-extra/elligator2"
-                 "gitlab.com/yawning/edwards25519-extra/h2c"
-                 "gitlab.com/yawning/edwards25519-extra/internal/montgomery"
-                 "gitlab.com/yawning/edwards25519-extra/vrf"))))
-           (replace 'check
-             (lambda arguments
-               (for-each
-                (lambda (directory)
-                  (apply (assoc-ref %standard-phases 'check)
-                         `(,@arguments #:import-path ,directory)))
-                (list
-                 "gitlab.com/yawning/edwards25519-extra/elligator2"
-                 "gitlab.com/yawning/edwards25519-extra/h2c"
-                 "gitlab.com/yawning/edwards25519-extra/internal/montgomery"
-                 "gitlab.com/yawning/edwards25519-extra/vrf"))))
-           (replace 'install
-             (lambda arguments
-               (for-each
-                (lambda (directory)
-                  (apply (assoc-ref %standard-phases 'install)
-                         `(,@arguments #:import-path ,directory)))
-                (list
-                 "gitlab.com/yawning/edwards25519-extra/elligator2"
-                 "gitlab.com/yawning/edwards25519-extra/h2c"
-                 "gitlab.com/yawning/edwards25519-extra/internal/montgomery"
-                 "gitlab.com/yawning/edwards25519-extra/vrf")))))))
-      (propagated-inputs (list go-golang-org-x-crypto
-                               go-filippo-io-edwards25519))
-      (home-page "https://gitlab.com/yawning/edwards25519-extra")
-      (synopsis "edwards25519-extra")
-      (description
-       "This package provides extensions to the Go standard library's Ed25519 and
-curve25519 implementations, primarily extracted from
-@@url{https://github.com/oasisprotocol/curve25519-voi,curve25519-voi}.  This
-package is intended for interoperability with the standard library and the
-@@url{https://filippo.io/edwards25519,edwards25519} package as much as possible.")
-      (license license:bsd-3))))
-
 (define-public go-github-com-bwesterb-go-ristretto
   (package
     (name "go-github-com-bwesterb-go-ristretto")
-- 
2.41.0

